ui/vty-lib/src/Ariadne/UI/Vty/Face.hs

```diff
@@ -74,349 +23,31 @@ data UiFeatures = UiFeatures
-  | UiDelBrokenWallets Text
-  deriving Eq
-
-----------------------------------------------------------------------------
--- UI commands
-----------------------------------------------------------------------------
-
--- | Commands issued by the UI widgets
-data UiCommand
-  = UiSelect [Word]
-  | UiBalance
-  | UiTxHistory
-  | UiSend UiSendArgs
-  | UiFee UiFeeArgs
-  | UiNewWallet UiNewWalletArgs
-  | UiNewAccount UiNewAccountArgs
-  | UiNewAddress UiNewAddressArgs
-  | UiRestoreWallet UiRestoreWalletArgs
-  | UiRename UiRenameArgs
-  | UiRemove
-  | UiExport
-  | UiKill Natural
-
-data UiSendOutput = UiSendOutput
-  { usoAddress :: !Text
-  , usoAmount :: !Text
-  }
-
-data UiSendArgs = UiSendArgs
-  { usaWalletIdx :: !(Maybe Word)
-  , usaAccounts :: ![Word32]
-  , usaOutputs :: [UiSendOutput]
-  , usaPassphrase :: !Text
-  }
-
-data UiFeeArgs = UiFeeArgs
-  { ufaWalletIdx :: !(Maybe Word)
-  , ufaAccounts :: ![Word32]
-  , ufaOutputs :: [UiSendOutput]
-  }
-
-data UiNewWalletArgs = UiNewWalletArgs
-  { unwaName :: !Text
-  , unwaPassphrase :: !Text
-  }
-
-data UiNewAccountArgs = UiNewAccountArgs
-  { unaaWalletIdx :: !(Maybe Word)
-  , unaaName :: !Text
-  }
-
-data UiNewAddressArgs = UiNewAddressArgs
-  { unadaWalletIdx :: !(Maybe Word)
-  , unadaAccountIdx :: !(Maybe Word)
-  }
-
-data UiRestoreWalletArgs = UiRestoreWalletArgs
-  { urwaName :: !Text
-  , urwaMnemonic :: !Text
-  , urwaPassphrase :: !Text
-  }
-
-data UiRenameArgs = UiRenameArgs
-  { uraName :: !Text
-  }
-
-----------------------------------------------------------------------------
--- UI command results
-----------------------------------------------------------------------------
-
--- | Results of commands issued by the UI widgets
-data UiCommandResult
-  = UiBalanceCommandResult UiBalanceCommandResult
-  | UiTxHistoryCommandResult UiTxHistoryCommandResult
-  | UiSendCommandResult UiSendCommandResult
-  | UiFeeCommandResult UiFeeCommandResult
-  | UiNewWalletCommandResult UiNewWalletCommandResult
-  | UiNewAccountCommandResult UiNewAccountCommandResult
-  | UiNewAddressCommandResult UiNewAddressCommandResult
-  | UiRestoreWalletCommandResult UiRestoreWalletCommandResult
-  | UiRenameCommandResult UiRenameCommandResult
-  | UiRemoveCommandResult UiRemoveCommandResult
-  | UiExportCommandResult UiExportCommandResult
-
-data UiBalanceCommandResult
-  = UiBalanceCommandSuccess Text
-  | UiBalanceCommandFailure Text
-
-data UiSendCommandResult
-  = UiSendCommandSuccess Text
-  | UiSendCommandFailure Text
-
-data UiFeeCommandResult
-  = UiFeeCommandSuccess Text
-  | UiFeeCommandFailure Text
-
-data UiTxHistoryRowPart = UiTxHistoryRowPart
-  { uthrpAddress :: Text
-  , uthrpAmount :: Text
-  }
-  deriving (Eq, Show)
-
-data UiTxHistoryRow = UiTxHistoryRow
-  { uthrId :: Text
-  , uthrTotal :: Text
-  , uthrFrom :: [UiTxHistoryRowPart]
-  , uthrTo :: [UiTxHistoryRowPart]
+data instance UiCurrency Vty
+  = UiCurrency
+  { getUiCurrency :: (Maybe Text)
```

Comment:
Please add a comment explaining what `Nothing` means. And why the fact that it can be `Nothing` belongs to this type, not the place where it's used (`Maybe UiCurrency`).
(I don't know how it's used, so it's not obvious to me whether it's a good idea or not)

Reply:
Also do we want to have something different in Qt?

Reply:
> I wasn't actually sure what Maybe is for here, it was in the original Face.hs so I left it as is. But as far as I can see, this value is actually never can be Nothing.

I suspect it might be needed in Disciplina. Could you please check? I am even quite sure this `Maybe` is for Disciplina, but whether it (`Maybe`) should be part of this type or applied externally depends on how it's used in Disciplina.

Reply:
My idea was for each backend to provide some kind of string identifier (`ada`, `dscp`, ...), which would later be used by a frontend to find a corresponding icon in resources, or use fallback text display otherwise.

This way, frontend normally doesn't care about what backend is actually used, but if it actually has some backend-specific resources bundled (e.g., currency icon), it can utilize them.

Reply:
Yeah, I believe a system like this is what I'm going to work on after I finish this task.

Reply:
Well, my primary concern is that it should be possible to use this new code in Disciplina.